viz.Rmd

```diff
@@ -85,82 +94,172 @@ making it easier for them to understand and remember your message quickly.
+(CO$_{\text{2}}$, in parts per million) 
+at the Mauna Loa research station in \index{Mauna Loa CO2} Hawaii 
+from 1959 onwards [@maunadata].
+For this book, we are going to focus on the last 40 years of the data set,
+1980-2020.
+
+**Question:** \index{question!visualization} 
+Does the concentration of atmospheric CO$_{\text{2}}$ change over time, 
+and are there any interesting patterns to note?
+
+```{r, echo = FALSE, warning = FALSE, message = FALSE}
+# convert year and month to date column
+# note: eventually move this to the data script
+library(lubridate)
+read_csv("data/mauna_loa.csv") |>
+  unite(col = "date_measured", year:month, remove = TRUE, sep = "-") |>
+  mutate(date_measured = ym(date_measured)) |>
+  select(-date_decimal) |>
+  filter(ppm > 0, date_measured > date("1980/01/01")) |>
+  #filter(ppm > 0, date_measured > interval(ymd("1980/01/01"), ymd("2021-01-01"))) |>
+  write_csv("data/mauna_loa_data.csv")
+```
 
-### The Mauna Loa CO2 data set 
- The [Mauna Loa CO2 data set](https://www.esrl.noaa.gov/gmd/ccgg/trends/data.html), curated by [Dr. Pieter Tans, NOAA/GML](https://www.esrl.noaa.gov/gmd/staff/Pieter.Tans/) and [Dr. Ralph Keeling, Scripps Institution of Oceanography](https://scrippsco2.ucsd.edu/)
-records the atmospheric concentration of carbon dioxide (CO2, in parts per million) at the Mauna Loa research station in \index{Mauna Loa CO2} Hawaii from 1959 onwards [@maunadata].
+To get started, we read in and take a look at the data:
 
-**Question:** \index{question!visualization} Does the concentration of atmospheric CO2 change over time, and are there any interesting patterns to note?
 ```{r 03-data-co2, warning=FALSE, message=FALSE}
 # mauna loa carbon dioxide data
-co2_df <- read_csv("data/mauna_loa.csv") |>
-  filter(ppm > 0, date_decimal < 2000)
+co2_df <- read_csv("data/mauna_loa_data.csv")
 co2_df
 ```
 
-Since we are investigating a relationship between two variables (CO2 concentration and date), a scatter plot is a good place to start. Scatter plots 
-show the data as individual points with `x` (horizontal axis) and `y` (vertical axis) coordinates. Here, we will use the decimal
- date as the `x` coordinate 
-and CO2 concentration as the `y` coordinate. When using the `ggplot2` package, we create the plot object with the `ggplot` function; there are 
-a few basic aspects of a plot that we need to specify:
+We see we have two columns available to us; `date_measured` and `ppm`. 
+The `date_measured` column holds the date the measurement was taken, 
+and is of type `date`.
+The `ppm` columns holds the value of $CO_2$ in parts per million 
+that was measured on each date, and is type `double`.
+
+> **Note:** `read_csv` was able to parse the `date_measured` column into the
+> `date` vector type because it was entered 
+> in the international standard date format, 
+> called ISO 8601, which lists dates as `year-month-day`.
+> `date` vectors are `double` vectors with special properties that allow 
```

Comment:
I think we should move the line : "date vectors are double vectors with special properties that allow them to handle dates correctly. For exam- ple, date type vectors allow functions like ggplot to treat them as numeric dates and not a character vectors, even though they contain non-numeric characters (e.g., - in the date_measured column in the co2_df data frame). " to the beginning of this note and then say "read_csv was able to parse..."